### ISSUE TITLE
Questions about errors in running the code provided in Supplementary information 

### ISSUE THREAD

**Issue by Shaoqing1117** [elakatos/CloneGrowthSimulation#1]
Hi,  had the opportunity to read this excellent article rentlyce and wanted to explore more details about this research. Having been impressed by the interesting predictions from the simulations, I decided to go through the codes but it didn't go well, I am pretty new to Julia, or coding, forgive if it's a dumb question.
When I tried to run the codes in the file IdealPopulation_Scaled.jl,  the fololowing error emerged(Julia version 0.5.2)，

 Julia .\IdealPopulation_Scaled.jl
WARNING: Method definition describe(AbstractArray) in module StatsBase at C:\Users\17263\.julia\v0.5\StatsBase\src\scalarstats.jl:560 overwritten in module DataFrames at C:\Users\17263\.julia\v0.5\DataFrames\src\abstractdataframe\abstractdataframe.jl:407.
Mutations processed for 100000 cells.
Mutations processed for 100000 cells.
ERROR: LoadError: MethodError: no method matching colon(::Int64, ::Distributions.Exponential{Float64})
Closest candidates are:
  colon{T<:Real}(::T<:Real, ::Any, !Matched::T<:Real) at range.jl:114
  colon{A<:Real,C<:Real}(::A<:Real, ::Any, !Matched::C<:Real) at range.jl:112
  colon{T}(::T, ::Any, !Matched::T) at range.jl:118
  ...
 in start_population(::Float64, ::Int64, ::Distributions.Exponential{Float64}, ::Bool, ::Float64) at C:\Users\17263\Downloads\CloneGrowthSimulation-v1.0 (1)\elakatos-CloneGrowthSimulation-09a38ec\IdealPopulation_Scaled.jl:73
 in birthdeath_neoep(::Int64, ::Float64, ::Float64, ::Float64, ::Distributions.Exponential{Float64}, ::Int64, ::Int64, ::Float64) at C:\Users\17263\Downloads\CloneGrowthSimulation-v1.0 (1)\elakatos-CloneGrowthSimulation-09a38ec\IdealPopulation_Scaled.jl:178
 in macro expansion; at C:\Users\17263\Downloads\CloneGrowthSimulation-v1.0 (1)\elakatos-CloneGrowthSimulation-09a38ec\IdealPopulation_Scaled.jl:203 [inlined]
 in anonymous at .\<missing>:?
 in include_from_node1(::String) at .\loading.jl:488
 in process_options(::Base.JLOptions) at .\client.jl:265
 in _start() at .\client.jl:321
while loading C:\Users\17263\Downloads\CloneGrowthSimulation-v1.0 (1)\elakatos-CloneGrowthSimulation-09a38ec\IdealPopulation_Scaled.jl, in expression starting on line 202

**Comment by elakatos** [comment]
Hi!

I'm very novice in Julia too, so this is probably due to some sub-optimal coding practice from me... :) I remember occasionally getting this issue, but it was quite random whether it would appear. Maybe you could try changing the iterator in line 184 (and after) of the original script (for i=1:100) to something different, like iter=1:100?
If the issue still persist, can you share the script you are running, so I can try to reproduce the issue?

Eszter

**Comment by hte123** [comment]
Hi! 
I'm also trying to reproduce some of the results in this interesting and inspiring research recently. I found that the bug lies in line 160 of IdealPopulation_Scaled.jl. I changed `cells, mutID, muts, nonimm, N = start_population(p, initial_mut, neoep_dist)` to `cells, mutID, muts, nonimm, N = start_population(p, neoep_dist, initial_mut)` , and then I can run the code without error. The issue is random because the block of code is run only when all tumor cells have died.
I hope it will be helpful!

Tianen He
